CMakeFuncs.cmake

```diff
@@ -0,0 +1,627 @@
+# This is a set of common functions and settings for rippled
```

Comment:
Why is it that I get away with the most simple build scripts, like:
https://github.com/vinniefalco/NuDB/blob/master/tools/CMakeLists.txt

...while other build scripts grow so large that they threaten to secede and form their own country?